pkg/insightstore/filepath.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightstore
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+// insight file paths according to the following format.
+//
+// insights
+//  ├─ project-id
+//    ├─ deployment-frequency
+//        ├─ project  # aggregated from all applications
+//            ├─ years.json
+//            ├─ 2020-01.json
+//            ├─ 2020-02.json
+//            ...
+//        ├─ app-id
+//            ├─ years.json
+//            ├─ 2020-01.json
+//            ├─ 2020-02.json
+//            ...
+func newYearlyFilePath(projectID string, metricsKind model.InsightMetricsKind, appID string) string {
+	metricsKindKebab := strings.ToLower(metricsKind.String())
+	return fmt.Sprintf("insights/%s/%s/%s/years.json", projectID, metricsKindKebab, appID)
+}
+
+func newMonthlyFilePath(projectID string, metricsKind model.InsightMetricsKind, appID string, month string) string {
```

Comment:
`makeChunkFilePath`
I think we should not use the `monthly` or `yearly` to avoid misunderstanding with the `MONTHLY` or `YEARLY` step.
Basically, we have a `years.json` file for storing all yearly data points, and multiple chunk files (`year-month.json`) for storing a block of data points.